specification/dimensions/pricingmodel.md

```diff
@@ -0,0 +1,39 @@
+# Pricing Model
+
+Pricing Model indicates the type of unit price applied to a charge at the time it was used or purchased. Pricing Model is commonly used to differentiate between charges at on-demand vs. reduced prices when investigating optimization opportunities, like for commitment-based discount coverage.
+
+The PricingModel column MUST be present. This column is of type String and MUST be one of the allowed values. PricingModel MUST NOT be null or empty if SkuPriceId is not null and MUST be null if SkuPriceId is null. PricingModel MUST be "Commitment-Based Discount" when CommitmentDiscountId is not null. PricingModel MUST be "Dynamic" when pricing is determined by the provider and may change over time, regardless of predetermined agreement pricing. PricingModel MUST be "Other" when a PricingModel is present but none of the current allowed values apply.
+
+## Column ID
+
+PricingModel
+
+## Display name
+
+Pricing Model
+
+## Description
+
+Type of unit price applied to a charge at the time it was used or purchased.
+
+## Content constraints
+
+|    Constraint   |      Value       |
+|:----------------|:-----------------|
+| Column required | True             |
+| Data type       | String           |
+| Allows nulls    | True             |
+| Value format    | list-of-values   |
+
+Allowed values:
+
+| Value                     | Description                                                                                                                                                                                                  |
+| :------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| On-Demand                 | Charges priced at the standard rate for the billing account. This includes any applicable negotiated discounts and volume/tiered pricing, but does not include dynamic or commitment-based discount pricing. |
+| Dynamic                   | Charges priced at a variable rate determined by the provider. This includes any product or service with a unit price the provider can change without notice, like interruptible or low priority resources.   |
+| Commitment-Based Discount | Charges with reduced prices due to a commitment-based discount specified by the Commitment Discount ID.                                                                                                      |
+| Other | Use when a PricingModel is present but none of the other allowed values apply. |
```

Comment:
After looking at this probably for way too long I have a proposal I would like the community to consider here. There is questions about the use of dynamic vs on-demand for Spot/Interruptible/Preemptive instances. There are examples of how this differs across CSPs. I am proposing we rename this dimension to Pricing Category with a future Pricing Subcategory to come.

The definition of values already here works ok when paired with some possible subcategories. E.g.

Description | Pricing Category | Pricing Subcategory | Charge Frequency | Commitment Discount ID
-- | -- | -- | -- | --
Purchase something as a one-time buy (cert, domain) | On-Demand | Standard | One-Time | NULL
Purchase something on a recurring basis (RI with partial upfront) | On-Demand | Standard | Recurring | NULL
Use something based on on-demand prices with no commitments (EBS volume usage) | On-Demand | Standard | Usage-Based | NULL
Use something based on tiered prices with no commitments (S3 storage) | On-Demand | Tiered | Usage-Based | NULL
Use something based on a commitment (EC2 with commitment) | Commitment-Based | Spend | Usage-Based | ID:123
Use a Spot instance | Dynamic | Spot | Usage-Based | NULL

I like the consistency this builds with category and subcategory. It also avoids the conflict in naming for PricingModels and PurchaseModels as this is just putting pricing into different categories that the FinOps practitioner is looking for. 

Some examples of how this can be used by the practitioner:

- In order to find usage that is priced at list price `ChargeCategory=Usage,PricingCategory=On-Demand`
- In order to find usage that is spot `ChargeCategory=Usage,PricingSubcategory=Spot`
- To work out how much of your usage has tiering `ChargeCategory=Usage,PricingCategory=On-Demand,PricingSubcategory=Tiered`

Reply:
I love this! I can imagine a lot of possibilities with PricingSubcategory to cover more scenarios that we weren't able to get in. I was debating using ChargeSubcategory for that (hence the null over there), but I do feel like this makes _much_ more sense, given these things are truly about "pricing" and not a generic explanation of the type of "usage".

Reply:
Simply: Yes!

Reply:
Tiering (aka volume-based discounts) is a pricing model where you get a built-in discount the more you use. You don't "select" a tier, but you do select a service that leverages tiering. You may not have a choice for that specific service, but there are likely other services that can be used that don't have tiering. More importantly, this also applies vice-versa. Let's say you use a DB that isn't tiered, then you see some DB charges are tiered and others aren't. As an optimization, you might move that untiered workload over to the tiered service to get more out of what you're spending. (This assumes all other things being equal, of course.)

Reply:
Turns out spot isn't a commercial thing at all. It's a general financial/stock term that refers to the price something can be immediately sold for "right now", noting that this price is always changing based on supply and demand. Sound familiar? 🙂 We should be fine without PricingType, but it's nice to have that option, if we do need it. I suspect anything like that would end up in a SKU column.

Reply:
> Side question about "Purchase something on a recurring basis (RI with partial upfront)" having PricingCategory=On-Demand. If this is about regular payment for reservations, I would see PricingCategory=Commitment-Based, so as that I can identify the Purchase of commitment with ChargeType=Purchase and PricingModel=CommitmentBased. Or should such identification be done in another way?

I like the idea of making commitment purchases easily identifiable, but if the purpose of this column is to identify how that purchase was "priced", then it wouldn't be commitment-based pricing. Yes, you're purchasing a disocunt, but that purchase isn't based on a commitment. (Unless it is, then it's a different story 😜) You can use CommitmentDiscountCategory for that, but you'd need to filter out nulls rather than filtering to a specific value.

Reply:
I like the concept - the fact that we always were coming back to discussions on PricingModel suggested that there was an issue with overloading it, so I think the sub category that further defines it makes sense to me. 

@sarahcloudpm Thoughts on this?

Reply:
Updated. We'll keep PricingSubcategory separate given the time. If anyone feels it's critical for 1.0 and we have enough people willing to close it quickly, we can bring it in. Please speak up in Slack 🙂

Reply:
Thanks Michael for all the explanations!

About Tiering, I would optimize the cost of a DB workload by moving it from one DB technology (SKU) to another if it makes the cost decrease. It may happen that the cost decreases when moving from a Tiered DB technology to a non-Tiered DB technology. Moreover, there are other Tiering pricing dimensions like for example Standard/Premium Tier in Azure DataBricks. A cost optimization is indeed to switch from Premium to Standard when it's not needed. In the end, that's why I have a feeling that it will be difficult for Pricing Model / Category & Sub Category to reflect different pricing dimensions like Tier-ing and it makes more sense to me to keep it only high-level, i.e. typically on-demand / commitment-based / interruptible.

Thanks about the explanation of spot! But then spot would also apply for AWS reservation marketplace? Moreover, does spot imply that the SKU can be interrupted (different SLA from non-spot)?

I am not very comfortable with identifying purchases of commitments with CommitmentDiscountCategory, because it would indeed apply in the theoretical (funny) case where the purchase is based on a commitment. BTW, in this case we would have two commit IDs: the one used for the purchase, and the one resulting from the purchase. Will provider ever introduce commitment-based purchases of commitments? In that case, I would introduce a PurchasingPricingCategory and a PurchasingCommitmentDiscountId. Indeed, I would really prefer to identify purchases and usages of commitments in the same way, based on PricingCategory.

Maybe I miss some background: feel free to ignore these comments, as the V1.0 deadline is coming soon.